pkg/planner/core/operator/logicalop/logical_selection.go

```diff
@@ -99,23 +100,26 @@ func (p *LogicalSelection) PredicatePushDown(predicates []expression.Expression,
 	exprCtx := p.SCtx().GetExprCtx()
 	stmtCtx := p.SCtx().GetSessionVars().StmtCtx
 	predicates = constraint.DeleteTrueExprs(exprCtx, stmtCtx, predicates)
-	p.Conditions = constraint.DeleteTrueExprs(exprCtx, stmtCtx, p.Conditions)
+	p.Conditions = utilfuncp.ApplyPredicateSimplification(p.SCtx(), p.Conditions, true)
 	var child base.LogicalPlan
 	var retConditions []expression.Expression
 	var originConditions []expression.Expression
 	canBePushDown, canNotBePushDown := splitSetGetVarFunc(p.Conditions)
 	originConditions = canBePushDown
 	retConditions, child = p.Children()[0].PredicatePushDown(append(canBePushDown, predicates...), opt)
 	retConditions = append(retConditions, canNotBePushDown...)
+	sctx := p.SCtx()
 	if len(retConditions) > 0 {
-		p.Conditions = expression.PropagateConstant(exprCtx, retConditions...)
+		p.Conditions = utilfuncp.ApplyPredicateSimplification(sctx, retConditions, false)
 		// Return table dual when filter is constant false or null.
 		dual := Conds2TableDual(p, p.Conditions)
 		if dual != nil {
 			AppendTableDualTraceStep(p, dual, p.Conditions, opt)
 			return nil, dual
 		}
 		return nil, p
+	} else if len(retConditions) == 0 {
```

Comment:
the if len(retConditions) > 0's else should be len(retConditions) <= 0?
i think we can remove the len(retConditions) == 0 here ?